app/org/sagebionetworks/bridge/models/schedules/ActivityScheduler.java

```diff
@@ -42,21 +41,36 @@ protected DateTime getScheduledTimeBasedOnEvent(ScheduleContext context) {
     protected void addScheduledActivityForAllTimes(List<ScheduledActivity> scheduledActivities, 
```

Comment:
Maybe in order to make this easier to reason about, we should have this method take in a LocalDate instead of a DateTime. For schedules without times, we can add midnight to make it a LocalDateTime. For schedules with, we use those times to make LocalDateTimes. Since tasks are scheduled using LocalDateTimes, this method and tasks work the same way, and we don't have to deal with timezones at all under this call chain.

Conceptually, this means that Interval and Cron scheduler can only be used to pick the day of the activity (which de facto true once we make this change), and in order to select the specific times of day, you need to specify the list of times.

This should be stable as long as the LocalDates we're passing in are stable. Which should be true if all events are stored in UTC and handled in UTC, which is true after this change.

There are only 2 uses of add...ForAllTimes() and only 3 uses of add...ForTime(), so it shouldn't be too time-consuming to refactor.